demos/shadow/shadow_demo_main/shadow_demo_main.c

```diff
@@ -211,6 +256,84 @@ static void updateDeltaHandler( MQTTPublishInfo_t * pPublishInfo );
  */
 static void updateAcceptedHandler( MQTTPublishInfo_t * pPublishInfo );
 
+/**
+ * @brief Process payload from `/delete/rejected` topic.
+ *
+ * This handler examines the rejected message to look for the reject reason code.
+ * If the reject reason code is `404`, an attempt was made to delete a shadow
+ * document which was not present yet. This is considered to be success for this
```

Comment:
What is the reason for treating a "rejected" response as successful for the demo? Would be good to mention the reasoning.

Reply:
The sentence above is the reason for it.